portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,90 @@
 
 package com.google.sps.servlets;
 
+import java.util.*;
 import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter; 
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public class DataServlet extends HttpServlet 
+{
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException 
+    {
+        Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService savedComments = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = savedComments.prepare(query);
+
+        List<Task> tasks = new ArrayList<>();
+
+        for (Entity entity : results.asIterable()) 
+        {
+            long id = entity.getKey().getId();
+            long timestamp = (long) entity.getProperty("timestamp");
+            String comment = (String) entity.getProperty("comment");
+            
+            Task task = new Task(id, timestamp, comment);
+            tasks.add(task);
+        }
+
+        Gson gson = new Gson();
+
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(tasks));
+    }
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException 
+    {
+        String comment = getParameter(request, "text-input", "");
+        long timestamp = System.currentTimeMillis();
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+        Entity taskEntity = new Entity("Task");
+        taskEntity.setProperty("comment", comment);
+        taskEntity.setProperty("timestamp", timestamp);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(taskEntity);
+
+        response.sendRedirect("/index.html");
+    }
+
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) 
+    {
+        String value = request.getParameter(name);
+        
+        if (value == null) 
+        {
+        return defaultValue;
+        }
+
+        return value;
```

Comment:
Read about and consider using the ternary operator here.

Reply:
I know how to use ternary operators, but I wasn't sure if I should use them because I thought it might be harder to read. I'll make the changes

Reply:
I agree ternary operator may not be easier to read here, think Herman isn't insisting on the use of it.